driver-core/src/main/com/mongodb/connection/netty/NettyStream.java

```diff
@@ -122,7 +124,7 @@ public void initChannel(final SocketChannel ch) throws Exception {
                     }
                     ch.pipeline().addFirst("ssl", new SslHandler(engine, false));
                 }
-                ch.pipeline().addLast("readTimeoutHandler", new ReadTimeoutHandler(settings.getReadTimeout(MILLISECONDS), MILLISECONDS));
+                ch.pipeline().addLast(READ_HANDLER_NAME, new ReadTimeoutHandler(settings.getReadTimeout(MILLISECONDS)));
```

Comment:
Why do the extra work if there is no read timeout configured?  Can we just not register a handler in that case?